packages/types/src/misc.ts

```diff
@@ -65,3 +65,8 @@ export interface WorkerLocation {
 export type Primitive = number | string | boolean | bigint | symbol | null | undefined;
 
 export type HttpHeaderValue = string | string[] | number | null;
+
+/**
+ * Type representing a semver version string
+ */
+export type VersionString = `${number}.${number}.${number}${`-${string}${string}` | ''}`;
```

Comment:
```suggestion
export type VersionString = `${number}.${number}.${number}${`-${string}` | ''}`;
```

do we need the double string there? 🤔

Reply:
the double string `{string}{string}` ensures that there's no empty preview string (e.g. `8.0.0-`). I might have to ditch this type entirely if I don't get the TS3.8 tests to pass I'm afraid 🙃

Reply:
ah, I see. if we have to ditch it, I guess also fine!